tests/integration/testlib/nmplugin.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+import re
+
+from . import cmdlib
+
+
+SYSTEMCTL_RESTART_NM_CMD = ("systemctl", "restart", "NetworkManager")
+
+
+@contextmanager
+def disable_nm_plugin(plugin):
+    ret, rpm_output, _ = cmdlib.exec_cmd(
+        ("rpm", "-ql", f"NetworkManager-{plugin}")
+    )
+    if not ret:
+        lib_path = re.search(r"[A-Za-z0-9-/.]*.so", rpm_output).group()
+        with mount_devnull_to_path(lib_path):
+            yield
+    else:
+        yield
+
+
+@contextmanager
+def mount_devnull_to_path(lib_path):
+    cmdlib.exec_cmd(("mount", "--bind", "/dev/null", lib_path))
+    try:
+        with nm_service_restart():
+            try:
+                yield
+            finally:
+                cmdlib.exec_cmd(("umount", lib_path))
+    except Exception:
```

Comment:
What kind of exceptions are expected here? in case it is not possible to execute commands (which is the only thing that `nm_service_restart()` does, it should also not be possible to recover from this to execute just another command. Therefore this might be a little bit of overkill. But maybe I just looked too long at this code.

Reply:
I think you are right. This try: except: is a little bit overkill but I am not completely against it so I am not going to fight for it.

Reply:
Actually, when nm_service_restart would raise an exception if the exception handling becomes more meaningful.

Reply:
Done, thanks!